### PR TITLE
Update Go to v1.18 and Add Generic Set data structure

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/gofrog#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
- [x] I labeled this pull request with one of the following: 'breaking change', 'new feature', 'bug', or 'ignore for release'

---
- Update gofrog to Go v1.18
- Add Generic Set data structure

### CODE REVIEW

File: datastructures/set_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	set.Add(5)
+	set.Add(7)
+
+	return set
+}
+
+func TestSetExistsAndAdd(t *testing.T) {
+	testSet := generateNewSetWithData()
+	assert.True(t, true, testSet.Exists(3))
+	assert.False(t, false, testSet.Exists(7))
```
Comment (yahavi) [inline datastructures/set_test.go]: Please notice that the second parameter of this assertion is the expected value. The third one is optional.
```suggestion
	assert.True(t, testSet.Exists(3))
	assert.False(t, testSet.Exists(4))
}
```

File: datastructures/set_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func TestSetRemove(t *testing.T) {
+	testSet := generateNewSetWithData()
+	assert.NoError(t, testSet.Remove(5))
+	assert.Equal(t, testSet.Size(), 2)
+	assert.False(t, false, testSet.Exists(5))
```
Comment (yahavi) [inline datastructures/set_test.go]:
```suggestion
	assert.False(t, testSet.Exists(5))
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Looks good @omerzi!
Please see my inline comments.

File: go.mod [hunk 3]
```diff
@@ -5,4 +5,10 @@ require (
...
 )
 
-go 1.13
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+)
+
+go 1.18
```
Comment (sverdlov93) [inline go.mod]: github.com/pkg/errors can be updated to 0.9.1 and also move 'go 1.18' line to the top of the page

File: .github/workflows/frogbot-fix-go.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v2
```
Comment (sverdlov93) [inline .github/workflows/frogbot-fix-go.yml]:
```suggestion
        uses: actions/setup-go@v3
```

**Pull request merged**